include/fuse_common.h

```diff
@@ -593,10 +610,21 @@ struct fuse_conn_info {
 	 */
 	unsigned time_gran;
 
+	/**
+	 * When FUSE_CAP_PASSTHROUGH is enabled, this is the maximum allowed
+	 * stacking depth of the backing files. The default is 0, meaning that
+	 * the backing files cannot be on a stacked filesystem.
+	 * Set this to 1 if backing files are expected to be on a stacked
+	 * filesystem, such as overlayfs or another FUSE passthrough.
+	 * In current kernel, FILESYSTEM_MAX_STACK_DEPTH is 2, which includes
+	 * the FUSE layer, so max_backing_stack_depth > 1 is not supported.
```

Comment:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/fuse/inode.c?h=v6.9-rc2#n1326

If I'm reading this correctly... that is not correct. It isn't if max > 1 but max > 2. The kernel check is `arg->max_stack_depth <= FILESYSTEM_MAX_STACK_DEPTH`

Reply:
Good point. It's important to get this exactly right if you are eg running FUSE inside Docker container so you are already on top of overlayfs. The limit of 2 is *really* low.

Reply:
I guess it relates to `In current kernel, FILESYSTEM_MAX_STACK_DEPTH is 2, which includes
	 * the FUSE layer, so max_backing_stack_depth > 1 is not supported.`

Reply:
But the literal check in FUSE is <= 2. Unless there is an additional check somewhere being done being referenced. In which case why would FUSE be checking?

Reply:
I wonder if it would make sense to email the person who implemented the kernel feature and kindly request them to review the libfuse counterpart.

Reply:
I think I'd prefer a hybrid approach given that making it that binary choice would be awkward to undo down the road if it did change. Maybe just expand the docs to include that information? Described not too differently from what you said here.

Reply:
@trapexit Sorry for my late reply, you might have found it already, if not, see in `do_init` 
```
if (se->conn.want & FUSE_CAP_PASSTHROUGH) {
		outargflags |= FUSE_PASSTHROUGH;
		/*
		 * outarg.max_stack_depth includes the fuse stack layer,
		 * so it is one more than max_backing_stack_depth.
		 */
		outarg.max_stack_depth = se->conn.max_backing_stack_depth + 1;```

Reply:
ok, added FUSE_BACKING_STACKED_{OVER,UNDER} macros (not enum) and explained them in documentation